gramineproject/gramine
# [LibOS,PAL] Introduce `etc` emulation (currently only 'resolv.conf')

## Description of the changes 

The first `etc` passthrough file is the `/etc/resolv.conf`. Gramine reads the option to global PAL state, and LibOS uses it to create a pseudo file with its content.

In case of `nameserver` we support only IPv4 in canonical form and with skipped zeros (eg. 8.8, 8, 8.8.8), and IPv6 in abbreviations (`::1`, `ff::1:2`) and full IP addresses.
In case of `search` we expects a valid hostname.

## Issue

https://github.com/gramineproject/gramine/issues/689

## Next step

* Add support for `hostname`

## How to test this PR?

Check if `/etc/resolv.conf` is exists with the valid content.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/889)
